### PR TITLE
Rename drawcalls.index to drawcalls.indexOffset

### PR DESCRIPTION
**Pull request by benaadams**

### CODE REVIEW

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -1094,7 +1094,7 @@ THREE.WebGLRenderer = function ( parameters ) {
 
 				for ( var i = 0, il = offsets.length; i < il; i ++ ) {
 
-					var startIndex = offsets[ i ].index;
+					var startIndex = offsets[ i ].indexOffset;
```
Comment (WestLangley) [inline src/renderers/WebGLRenderer.js]: So, do you think that `startIndex` should be renamed to `indexOffset` ? I do.

File: src/objects/Line.js [hunk 2]
```diff
@@ -69,7 +69,7 @@ THREE.Line.prototype.raycast = ( function () {
 
 				if ( offsets.length === 0 ) {
 
-					offsets = [ { start: 0, count: indices.length, index: 0 } ];
+					offsets = [{ start: 0, count: indices.length, indexOffset: 0 }];
```
Comment (WestLangley) [inline src/objects/Line.js]: offsets = [ { start: 0, count: indices.length, indexOffset: 0 } ];

File: src/core/BufferGeometry.js [hunk 3]
```diff
@@ -1018,7 +1018,7 @@ THREE.BufferGeometry.prototype = {
 
 				vertexMap[ vid ] = new_vid;
 				revVertexMap[ new_vid ] = vid;
-				sortedIndices[ indexPtr ++ ] = new_vid - offset.index; //XXX overflows at 16bit
+				sortedIndices[indexPtr++] = new_vid - offset.indexOffset; //XXX overflows at 16bit
```
Comment (WestLangley) [inline src/core/BufferGeometry.js]: sortedIndices[ indexPtr ++ ]

This is what mrdoobapproves would do.

File: src/core/BufferGeometry.js [hunk 4]
```diff
@@ -995,15 +995,15 @@ THREE.BufferGeometry.prototype = {
...
+			if ( faceMax > ( offset.indexOffset + size ) ) {
+				var new_offset = { start: indexPtr, count: 0, indexOffset: vertexPtr };
 				offsets.push( new_offset );
 				offset = new_offset;
 
 				//Re-evaluate reused vertices in light of new offset.
 				for ( var v = 0; v < 6; v += 2 ) {
 					var new_vid = faceVertices[ v + 1 ];
-					if ( new_vid > - 1 && new_vid < offset.index )
+					if ( new_vid > -1 && new_vid < offset.indexOffset )
```
Comment (WestLangley) [inline src/core/BufferGeometry.js]: if ( new_vid > - 1 && new_vid < offset.indexOffset )

OK. I'll stop.

File: src/core/BufferGeometry.js (hunk 3 above)
Comment (benaadams) [inline src/core/BufferGeometry.js]: My editor gets very annoyed with square brackets; is ok with round brackets but it eats square brackets spaces. So I have to edit then undo to restore formatting - don't always catch it :(

**Comment by WestLangley** [comment]
Thanks! I think this PR makes things clearer.

A remaining issue is the use of `drawcalls` and `offsets` to mean the same thing. We need to pick one and use it. I assume the intent was to use `drawcalls`.

**Comment by benaadams** [comment]
> A remaining issue is the use of drawcalls and offsets to mean the same thing. We need to pick one and use it. I assume the intent was to use drawcalls.

Agree, but I kept doing bad things when I put too many change types in one PR. So limiting them to one change type so @mrdoob has less fixing to do per PR ;-)

**Comment by WestLangley** [comment]
Oh yes, I agree. I was just thinking out loud. : - )

File: src/core/BufferGeometry.js (hunk 4 above)
Comment (mrdoob) [inline src/core/BufferGeometry.js]: Haha

**Comment by mrdoob** [comment]
I wonder if maybe `startIndex` is a better name than `indexOffset`...

Some more thinking out loud... I was starting to consider turning all this into `BufferDrawcall` (like `BufferAttribute`) and then moving to `.addDrawcall()`. With that done, then I think it could be cool if we added a `.material` property so we could have different materials in the same geometry (and somehow port all the `MeshFaceMaterial` stuff.

Having said that, I'm still unsure about the `drawcalls` and/or `offsets` naming...

**Comment by WestLangley** [comment]
> I wonder if maybe startIndex is a better name than indexOffset...

True, it is called `startIndex` in the renderer. I found that confusing, as I understood it to be an offset that is added to the index array value. Hence. `indexOffset`.

But now as I type this, I can see that one could consider it a "start index", with the index array being the "offset" -- which, of course, is why they are called offsets.

**Comment by benaadams** [comment]
Closing

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6498

**Pull request closed without merging**